src/java/voldemort/store/readonly/ReadOnlyFileDiffer.java

```diff
@@ -0,0 +1,383 @@
+package voldemort.store.readonly;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import voldemort.utils.ByteUtils;
+import voldemort.utils.EventThrottler;
+import voldemort.utils.Pair;
+
+
+/**
+ * ReadOnlyFileDiffer is used to compare between different versions of specific store and give the number of how many
+ * records are added/deleted/updated/same.
+ *
+ * N.B.: This tool only works with the Read-Only V2 format.
+ */
+public class ReadOnlyFileDiffer {
+    public final static int ADD = 0;
+    public final static int DELETE = 1;
+    public final static int UPDATE = 2;
+    public final static int SAME = 3;
+
+    public static void main(String[] args) {
+        //Initialize options of command.
+        Options options = new Options();
+        Option pathOption = new Option("p", "path", true, "Path of Voldemort readonly data dir");
+        pathOption.setRequired(true);
+        Option storeOption = new Option("s", "store", true, "Store name");
+        storeOption.setRequired(true);
+        Option oldOption = new Option("o", "oldVersion", true, "Old version number");
+        oldOption.setRequired(true);
+        Option newOption = new Option("n", "newVersion", true, "New version number");
+        newOption.setRequired(true);
+        options.addOption(pathOption);
+        options.addOption(storeOption);
+        options.addOption(oldOption);
+        options.addOption(newOption);
+        options.addOption("r", "rate", true, "Max rate of reading bytes from disk per second.");
+        options.addOption("h", "help", false, "Print usage");
+
+        CommandLineParser parser = new BasicParser();
+        CommandLine cmd;
+        try {
+            cmd = parser.parse(options, args);
+            //Print help information.
+            if (cmd.hasOption("h")) {
+                HelpFormatter formatter = new HelpFormatter();
+                formatter.printHelp("ReadOnlyFileDiffer", options);
+                return;
+            }
+
+            String path = cmd.getOptionValue("p");
```

Comment:
Can you make sure that if the parameters are not passed in correctly, it will print the help message and error out ? AFAIK, Options setRequired ensures that you are parameters are not flags and requires value, but you can skip the entire parameter and it does not notify you.